### PR TITLE
fix(errors): Rename Exception to Error; remove from public API

### PR DESCRIPTION
**Pull request by mhevery**
**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

BREAKING CHANGE:

Exceptions are no longer part of the public API. We don't expect that anyone should be referring to the Exception types.

### CODE REVIEW

File: modules/@angular/core/src/error_handler.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+ *   providers: [{provide: ErrorHandler, useClass: MyExceptionHandler}]
+ * })
+ * class MyModule {}
+ * ```
+ * @stable
+ */
+export class ErrorHandler {
+  constructor(private _logger: any, private _rethrowError: boolean = true) {}
+
+  call(error: any): void {
```
Comment (IgorMinar) [inline modules/@angular/core/src/error_handler.ts]: handleError

Comment (mhevery) [inline modules/@angular/core/src/error_handler.ts]: done

**Comment by mhevery** [comment]
@vicb Please use http://cr/131337806

**Comment by vicb** [comment]
merged via 7c07bff + previous commit

**Pull request closed without merging**